.github/workflows/docker.yaml

```diff
@@ -0,0 +1,419 @@
+        run: ${{ inputs.update_buildlog_command }}
+
+  # This job pushes the image manifest, AKA multi-arch image.
+  publish:
+    needs: [docker, image]
+    runs-on: ${{ inputs.other_jobs_runs_on }}
+    strategy:
+      matrix:
+        # Each entry in image is a space separated list of full image names with tags. We'll append "-$ARCH" to each one. The first
+        # in each list is expected to be the most canonical.
+        image: "${{ fromJSON(needs.image.outputs.images) }}"
+    env: # Needed for custom commands to access these variables at runtime
+      TAGS: ${{ needs.image.outputs.tags }}
+      ARCHS: ${{ inputs.archs }}
+      LONGTAG: ${{ needs.image.outputs.longtag }}
+      DOCKER_IMAGE: ${{ inputs.image_name }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Either the workflow_dispatch ref, or the prerelease tag, whichever is set.
+          ref: ${{ github.event.inputs.ref }}${{ github.event.release.tag_name }}
+      - name: Decrypt gcloud credentials
+        uses: IronCoreLabs/ironhide-actions/decrypt@v3
+        with:
+          keys: ${{ secrets.IRONHIDE_KEYS }}
+          input: .github/gcloud-auth.json.iron
+      - name: Configure gcloud
+        run: |
+          gcloud auth activate-service-account --key-file .github/gcloud-auth.json
+          gcloud auth configure-docker --quiet
+      - name: Get canonical image name
+        id: name
+        run: echo "canonical=$(echo ${{ matrix.image }} | sed 's/ .*//')" >> "$GITHUB_OUTPUT"
+      - name: Pull images (Default)
+        if: inputs.pull_images_command == ''
+        run: |
+          for ARCH in ${{ inputs.archs }} ; do
+            docker pull "${{ steps.name.outputs.canonical }}-${ARCH}"
+          done
+      - name: Pull images (Overridden)
+        if: inputs.pull_images_command != ''
+        run: ${{ inputs.pull_images_command }}
+      - name: Create and push manifest (Default)
+        if: inputs.create_and_push_manifest_command == ''
+        run: |
+          for I in ${{ matrix.image }} ; do
+            for ARCH in ${{ inputs.archs }} ; do
+              echo "${{ steps.name.outputs.canonical }}-${ARCH}"
+            done | \
+            xargs docker manifest create "${I}"
+            docker manifest push "${I}"
+          done
+      - name: Create and push manifest (Overridden)
+        if: inputs.create_and_push_manifest_command != ''
+        run: ${{ inputs.create_and_push_manifest_command }}
+
+  #  Trigger deployment.
+  trigger-deploy:
+    needs: [publish, image]
+    runs-on: ${{ inputs.other_jobs_runs_on }}
+    env:
+      IMAGE_NAME: ${{ needs.image.outputs.name }} # TODO: TSP expects this but it doesn't exist
+      IMAGE_TAGS: ${{ needs.image.outputs.tags }}
+      LAST_PR: ${{ needs.image.outputs.last_pr }} # TODO: TSP expects this but it doesn't exist
```

Comment:
Are these TODOs valid still? Were they ported over?

Reply:
They're new. Colt and I are talking about how to address them and I think we decided we can tear some stuff out, removing them altogether